services/users-ms/lambdas/findUser.js

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable no-console */
+import to from 'await-to-js';
+
+import config from '../../../config';
+import * as dynamoDb from '../../../libs/dynamoDb';
+import { putEvent } from '../../../libs/awsEventBridge';
+
+export async function main(event) {
+  const userDetail = event.detail;
+  const { personalNumber } = userDetail;
+
+  const [getUserError, userItem] = await to(getUser(personalNumber));
+  if (getUserError) {
+    return console.error('(users-ms: findUser) getUserError', getUserError);
+  }
+
+  if (userItem == undefined) {
+    const [emitEventError] = await to(emitEventUserNotFound(userDetail));
+    if (emitEventError) {
+      return console.error('(users-ms: findUser) emitEventError', emitEventError);
+    }
+    return console.log('User could not be found in the users table');
+  }
+
+  const [emitEventError] = await to(emitEventUserExists(userItem));
+  if (emitEventError) {
+    return console.error('(users-ms: findUser) emitEventError', emitEventError);
+  }
+  console.info('User was found in the users table.');
+  return true;
+}
+
+async function getUser(personalNumber) {
+  const params = {
+    TableName: config.users.tableName,
+    Key: {
+      personalNumber,
+    },
+  };
+
+  const [getError, getResult] = await to(dynamoDb.call('get', params));
+  if (getError) {
+    throw getError;
+  }
+
+  return getResult.Item;
+}
+
+async function emitEventUserNotFound(user) {
+  return putEvent(user, 'usersMsFindUserUnsucceeded', 'usersMs.findUser');
+}
+
+async function emitEventUserExists(user) {
+  return putEvent(user, 'usersMsFindUserSuccess', 'usersMs.findUser');
+}
```

Comment:
These functions feels very clear in their purpose but at the same time redundant, do not have any suggestions for other solutions at this moment but there is room for improvement i guess :)

Reply:
A thought on these functions might be to declare them in an object

```JS
const userEvents = {
  exists: async (user) => putEvent(user, ...),
  notFound: async (user) => putEvent(user, ...)
}

// then we use them like this
const eventResponse = userEvent.exists(user)
const eventResponse = userEvent.notFound(user)
```
You could also go further and do something like
```JS
// extend the incoming user object with functions/methods
const user = {
  ...user, 
  event: {
    send: (type) => putUserEvent(user, type) // don't now if there is a better way of referring to the user object here, but you get my intention
  }
}

async function putUserEvent(user, type) {
  const eventTypeCollection = {
   notFound: {...},
   exists: {...},
  }

  return await putEvent(...eventTypeCollection[type])
}

user.event.put('exists') 
```